lib/spack/spack/spec.py

```diff
@@ -3108,45 +3110,33 @@ def concretized(self, tests=False):
         clone.concretize(tests=tests)
         return clone
 
-    def flat_dependencies(self, **kwargs):
-        """Return a DependencyMap containing all of this spec's
-        dependencies with their constraints merged.
+    def flat_dependencies(self, disconnect: bool = False):
+        """Build DependencyMap of all of this spec's dependencies with their constraints merged.
 
-        If copy is True, returns merged copies of its dependencies
-        without modifying the spec it's called on.
-
-        If copy is False, clears this spec's dependencies and
-        returns them. This disconnects all dependency links including
-        transitive dependencies, except for concrete specs: if a spec
-        is concrete it will not be disconnected from its dependencies
-        (although a non-concrete spec with concrete dependencies will
-        be disconnected from those dependencies).
+        Arguments:
+            disconnect: if True, disconnect all dependents and dependencies among nodes in this
+                spec's DAG.
         """
-        copy = kwargs.get("copy", True)
-
         flat_deps = {}
-        try:
-            deptree = self.traverse(root=False)
-            for spec in deptree:
-                if spec.name not in flat_deps:
-                    if copy:
-                        spec = spec.copy(deps=False)
```

Comment:
This seems to be the only semantic change in the function. I tried to inspect closely, and I think it's ok to remove the copy. Do you recall why we added the copy here in the first place?

Reply:
It dates back to 2014 in 5f073ae2200. I guess it lasted not quite ten years :). `_dup` used to use `flat_dependencies()` to do its copying, but doesn't anymore. All remaining calls were passing `copy=False`, so I removed it.